### PR TITLE
Create runtime.Config struct with metric.Provider and WithMinimumReadMemStatsInterval() configuration options

### PR DESCRIPTION
**Pull request by jmacd**
Part of #213.

### CODE REVIEW

File: instrumentation/runtime/runtime.go [hunk 1]
```diff
@@ -26,18 +26,42 @@ import (
...
 // Runtime reports the work-in-progress conventional runtime metrics specified by OpenTelemetry
 type runtime struct {
-	meter    metric.Meter
-	interval time.Duration
+	meter              metric.Meter
+	minGCStatsInterval time.Duration
+}
+
+// Option supports configuring optional settings for runtime metrics.
+type Option func(*runtime)
```
Comment (evantorrie) [inline instrumentation/runtime/runtime.go:34]: Should we standardize the `Option/Apply` interface approach in `opentelemetry-go-contrib` the same as in `opentelemetry-go`, i.e. the approach [described here](https://github.com/open-telemetry/opentelemetry-go/issues/536)?

File: instrumentation/runtime/runtime.go [hunk 2]
```diff
@@ -26,18 +26,42 @@ import (
...
 // If the metric SDK attempts to observe MemStats-derived instruments more frequently than the
 // interval, a cached value will be used.
-func Start(meter metric.Meter, interval time.Duration) error {
+func Start(provider metric.Provider, opts ...Option) error {
 	r := &runtime{
-		meter:    meter,
-		interval: interval,
+		// TODO: How to set the instrumentation library version?
+		meter:              provider.Meter("runtime"),
+		minGCStatsInterval: DefaultMinimumGCStatsInterval,
```
Comment (evantorrie) [inline instrumentation/runtime/runtime.go:61]: Per https://github.com/open-telemetry/opentelemetry-go-contrib/issues/214 , I believe decision was to use a common instrumentation version across all implementations in this repo, but maybe that should be delayed to a later PR.

File: instrumentation/runtime/go.mod [hunk 3]
```diff
@@ -6,6 +6,7 @@ replace go.opentelemetry.io/contrib => ../..
 
 require (
 	github.com/stretchr/testify v1.6.1
+	go.opentelemetry.io/contrib v0.0.0-00010101000000-000000000000
```
Comment (MrAlias) [inline instrumentation/runtime/go.mod:9]:
```suggestion
	go.opentelemetry.io/contrib v0.10.1
```

File: instrumentation/runtime/runtime.go [hunk 4]
```diff
@@ -26,18 +26,42 @@ import (
...
+		}
+	}
 }
 
 // New returns Runtime, a structure for reporting Go runtime metrics
 // interval is used to limit how often to invoke Go runtime.ReadMemStats() to obtain metric data.
 // If the metric SDK attempts to observe MemStats-derived instruments more frequently than the
 // interval, a cached value will be used.
-func Start(meter metric.Meter, interval time.Duration) error {
+func Start(provider metric.Provider, opts ...Option) error {
```
Comment (MrAlias) [inline instrumentation/runtime/runtime.go:57]: The `Provider` should be passed as an option with the default, if none are passed, being the `global.MeterProvider` (https://github.com/open-telemetry/opentelemetry-go-contrib/issues/213).

File: instrumentation/runtime/doc.go [hunk 5]
```diff
@@ -1,3 +1,17 @@
+// Copyright The OpenTelemetry Authors
```
Comment (MrAlias) [inline instrumentation/runtime/doc.go:1]: Yikes, we need to add a similar check to the build to catch these...

Comment (MrAlias) [reply instrumentation/runtime/doc.go:1]: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/227

File: instrumentation/runtime/go.mod (hunk 3 above)
Comment (jmacd) [reply instrumentation/runtime/go.mod]: Done.

File: instrumentation/runtime/runtime.go (hunk 1 above)
Comment (jmacd) [reply instrumentation/runtime/runtime.go]: Fixed.

File: instrumentation/runtime/runtime.go [hunk 6]
```diff
@@ -26,18 +26,42 @@ import (
...
+// frequently.  `d` values less than 0 will be disregarded silently.
+func WithMinimumGCStatsInterval(d time.Duration) Option {
+	return func(r *runtime) {
+		if d >= 0 {
+			r.minGCStatsInterval = d
+		}
+	}
 }
 
 // New returns Runtime, a structure for reporting Go runtime metrics
```
[the remark this answers is not in the document]
Comment (jmacd) [reply instrumentation/runtime/runtime.go]: Updated.

File: instrumentation/runtime/runtime.go (hunk 4 above)
Comment (jmacd) [reply instrumentation/runtime/runtime.go]: Fixed.

File: instrumentation/runtime/runtime.go (hunk 2 above)
Comment (jmacd) [reply instrumentation/runtime/runtime.go]: Thanks for #225.

File: instrumentation/runtime/runtime.go [hunk 7]
```diff
@@ -20,48 +20,100 @@ import (
...
+	if c.MeterProvider == nil {
+		c.MeterProvider = global.MeterProvider()
+	}
+	r := &runtime{
+		meter: c.MeterProvider.Meter(
+			// TODO: should library names be qualified?
+			// e.g., contrib/runtime?
+			"runtime",
+			// TODO(#225): set the instrumentation library version
+			// metric.WithInstrumentationVersion(contrib.SemVersion())
```
Comment (evantorrie) [inline instrumentation/runtime/runtime.go:114]: #225 is merged to mainline, so should be able to uncomment this now.

Comment (jmacd) [reply instrumentation/runtime/runtime.go]: Done.

**Comment by jmacd** [comment]
From a terminology perspective, there's one lingering concern.  We've written `WithMeterProvider()` for a class named `metric.Provider` over and over again. I wonder if we should rename `metric.Provider` to `metric.MeterProvider`, but such a rename might cause more trouble than it's worth.

**Comment by MrAlias** [comment]
> From a terminology perspective, there's one lingering concern. We've written `WithMeterProvider()` for a class named `metric.Provider` over and over again. I wonder if we should rename `metric.Provider` to `metric.MeterProvider`, but such a rename might cause more trouble than it's worth.

I've been passively thinking of needing this kind of rename if we wanted to move the provider to the top level (alongside a `TraceProvider`). Sounds reasonable to me.

**Comment by jmacd** [comment]
> I've been passively thinking of needing this kind of rename if we wanted to move the provider to the top level (alongside a TraceProvider).

 I like this idea!

File: instrumentation/runtime/runtime.go [hunk 8]
```diff
@@ -20,24 +20,100 @@ import (
...
+	if c.MeterProvider == nil {
+		c.MeterProvider = global.MeterProvider()
+	}
 	r := &runtime{
-		meter:    meter,
-		interval: interval,
+		meter: c.MeterProvider.Meter(
+			// TODO: should library names be qualified?
+			// e.g., contrib/runtime?
+			"runtime",
```
Comment (MrAlias) [inline instrumentation/runtime/runtime.go:113]: Yeah, this should be fully qualified:

```suggestion
			// TODO: should library names be qualified?
			// e.g., contrib/runtime?
			"go.opentelemetry.io/contrib/instrumentation/runtime",
```

We can address this in a subsequent PR though.

Comment (MrAlias) [reply instrumentation/runtime/runtime.go:113]: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/240

**Pull request merged**